search_queries_interval_rules_filter.go

```diff
@@ -0,0 +1,142 @@
+package elastic
+
+type FilterRule struct {
+	after          Query
+	before         Query
+	containedBy    Query
+	containing     Query
+	overlapping    Query
+	notContainedBy Query
+	notContaining  Query
+	notOverlapping Query
+	script         *Script
+}
+
+var _ IntervalRule = &FilterRule{}
+
+func NewFilterRule() *FilterRule {
+	return &FilterRule{}
+}
+
+func (r *FilterRule) After(after Query) *FilterRule {
+	r.after = after
+	return r
+}
+func (r *FilterRule) Before(before Query) *FilterRule {
+	r.before = before
+	return r
+}
+func (r *FilterRule) ContainedBy(containedBy Query) *FilterRule {
+	r.containedBy = containedBy
+	return r
+}
+func (r *FilterRule) Containing(containing Query) *FilterRule {
+	r.containing = containing
+	return r
+}
+func (r *FilterRule) Overlapping(overlapping Query) *FilterRule {
+	r.overlapping = overlapping
+	return r
+}
+func (r *FilterRule) NotContainedBy(notContainedBy Query) *FilterRule {
+	r.notContainedBy = notContainedBy
+	return r
+}
+func (r *FilterRule) NotContaining(notContaining Query) *FilterRule {
+	r.notContaining = notContaining
+	return r
+}
+func (r *FilterRule) NotOverlapping(notOverlapping Query) *FilterRule {
+	r.notOverlapping = notOverlapping
+	return r
+}
+func (r *FilterRule) Script(script *Script) *FilterRule {
+	r.script = script
+	return r
+}
+
+// Source returns JSON for the function score query.
+func (r *FilterRule) Source() (interface{}, error) {
+	source := make(map[string]interface{})
+
+	if r.before != nil {
+		src, err := r.before.Source()
+		if err != nil {
+			return nil, err
+		}
+		source["before"] = src
+	}
+
+	if r.after != nil {
+		src, err := r.after.Source()
+		if err != nil {
+			return nil, err
+		}
+		source["after"] = src
+	}
+
+	if r.containedBy != nil {
+		src, err := r.containedBy.Source()
+		if err != nil {
+			return nil, err
+		}
+		source["contained_by"] = src
+	}
+
+	if r.containing != nil {
+		src, err := r.containing.Source()
+		if err != nil {
+			return nil, err
+		}
+		source["containing"] = src
+	}
+
+	if r.overlapping != nil {
+		src, err := r.overlapping.Source()
+		if err != nil {
+			return nil, err
+		}
+		source["overlapping"] = src
+	}
+
+	if r.notContainedBy != nil {
+		src, err := r.notContainedBy.Source()
+		if err != nil {
+			return nil, err
+		}
+		source["not_contained_by"] = src
+	}
+
+	if r.notContaining != nil {
+		src, err := r.notContaining.Source()
+		if err != nil {
+			return nil, err
+		}
+		source["not_containing"] = src
+	}
+
+	if r.notOverlapping != nil {
+		src, err := r.notOverlapping.Source()
+		if err != nil {
+			return nil, err
+		}
+		source["not_overlapping"] = src
+	}
+
+	if r.script != nil {
+		src, err := r.script.Source()
+		if err != nil {
+			return nil, err
+		}
+		source["script"] = src
+	}
+
+	// todo: not so clear from docs, if filter can be a top-level rule, and so
+	// 	if it does need a wrapper map[string]interface{}{"filter": ..} like other rules do
```

Comment:
Always use the Java source as the ultimate reference. The online docs are pretty good, but targeted to a broader understanding of the concepts, not hard facts.